src/index.d.ts

```diff
@@ -0,0 +1,825 @@
+        optSet(key: "punctuation-wrap" | "punctuation_wrap" | "PunctuationWrap", value: boolean): void;
+        /**
+         * merge-emphasis
+         */
+        optGet(key: "merge-emphasis" | "merge_emphasis" | "MergeEmphasis"): boolean;
+        optSet(key: "merge-emphasis" | "merge_emphasis" | "MergeEmphasis", value: boolean): void;
+        /**
+         * merge-divs
+         */
+        optGet(key: "merge-divs" | "merge_divs" | "MergeDivs"): "no" | "yes" | "auto";
+        optSet(key: "merge-divs" | "merge_divs" | "MergeDivs", value: "no" | "yes" | "auto"): void;
+        /**
+         * decorate-inferred-ul
+         */
+        optGet(key: "decorate-inferred-ul" | "decorate_inferred_ul" | "DecorateInferredUl"): boolean;
+        optSet(key: "decorate-inferred-ul" | "decorate_inferred_ul" | "DecorateInferredUl", value: boolean): void;
+        /**
+         * preserve-entities
+         */
+        optGet(key: "preserve-entities" | "preserve_entities" | "PreserveEntities"): boolean;
+        optSet(key: "preserve-entities" | "preserve_entities" | "PreserveEntities", value: boolean): void;
+        /**
+         * sort-attributes
+         */
+        optGet(key: "sort-attributes" | "sort_attributes" | "SortAttributes"): "none" | "alpha";
+        optSet(key: "sort-attributes" | "sort_attributes" | "SortAttributes", value: "none" | "alpha"): void;
+        /**
+         * merge-spans
+         */
+        optGet(key: "merge-spans" | "merge_spans" | "MergeSpans"): "no" | "yes" | "auto";
+        optSet(key: "merge-spans" | "merge_spans" | "MergeSpans", value: "no" | "yes" | "auto"): void;
+        /**
+         * anchor-as-name
+         */
+        optGet(key: "anchor-as-name" | "anchor_as_name" | "AnchorAsName"): boolean;
+        optSet(key: "anchor-as-name" | "anchor_as_name" | "AnchorAsName", value: boolean): void;
+        /**
+         * indent-with-tabs
+         */
+        optGet(key: "indent-with-tabs" | "indent_with_tabs" | "IndentWithTabs"): boolean;
+        optSet(key: "indent-with-tabs" | "indent_with_tabs" | "IndentWithTabs", value: boolean): void;
+        /**
+         * skip-nested
+         */
+        optGet(key: "skip-nested" | "skip_nested" | "SkipNested"): boolean;
+        optSet(key: "skip-nested" | "skip_nested" | "SkipNested", value: boolean): void;
+        /**
+         * strict-tags-attributes
+         */
+        optGet(key: "strict-tags-attributes" | "strict_tags_attributes" | "StrictTagsAttributes"): boolean;
+        optSet(key: "strict-tags-attributes" | "strict_tags_attributes" | "StrictTagsAttributes", value: boolean): void;
+        /**
+         * escape-scripts
+         */
+        optGet(key: "escape-scripts" | "escape_scripts" | "EscapeScripts"): boolean;
+        optSet(key: "escape-scripts" | "escape_scripts" | "EscapeScripts", value: boolean): void;
+    }
+
+    interface OptionDict {
+        indent_spaces?: number;
+        IndentSpaces?: number;
```

Comment:
If you specify the dict like this, people may conceivably pass conflicting options to the constructor. I'd rather use a white lie and omit all the camel-cased names. That helps keep the sources consistent and avoids conflicts.